pkg/admission/mutator/shoot.go

```diff
@@ -55,6 +56,28 @@ func (s *shoot) Mutate(ctx context.Context, new, old client.Object) error {
 		return fmt.Errorf("wrong object type %T", new)
 	}
 
+	// skip if shoot is in restore or migration phase
+	if wasShootRescheduledToNewSeed(shoot) {
```

Comment:
Can you add unit tests for the newly introduced checks? It seems that there are no unit tests for this webhook.